app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -7,6 +7,24 @@ import type {
 import type { KeyringController } from '@metamask/keyring-controller';
 import { SNAP_MANAGE_ACCOUNTS_CONFIRMATION_TYPES } from '../../../../shared/constants/app';
 import { t } from '../../translate';
+import MetamaskController from '../../metamask-controller';
+
+/**
+ * Get the addresses of the accounts managed by a given Snap.
+ *
+ * @param controller - Instance of the MetaMask Controller.
+ * @param snapId - Snap ID to get accounts for.
+ * @returns The addresses of the accounts.
+ */
+export const getAccountsBySnapId = async (
+  controller: MetamaskController,
+  snapId: string,
+) => {
+  const snapKeyring: SnapKeyring =
+    (await controller.getSnapKeyring()) as SnapKeyring;
```

Comment:
This type cast is needed since the base controller is untyped and the the return type of `getSnapKeyring` is unknown. This could cause errors but i'd argue that throwing an error if we pass in the wrong controller is probably a good thing.

Reply:
I was wrong, by adding the return type in the jsdoc here

```
  /**
   * Initialize the snap keyring if it is not present.
   *
   * @returns {SnapKeyring}
   */
  async getSnapKeyring() {
    let [snapKeyring] = this.coreKeyringController.getKeyringsByType(
      KeyringType.snap,
    );
    if (!snapKeyring) {
      snapKeyring = await this.coreKeyringController.addNewKeyring(
        KeyringType.snap,
      );
    }
    return snapKeyring;
  }
  ```
  
  

Our TS file is able to infer the return type and avoid the cast.